src/Databases/DatabaseMemory.cpp

```diff
@@ -132,15 +193,24 @@ UUID DatabaseMemory::tryGetTableUUID(const String & table_name) const
     return UUIDHelpers::Nil;
 }
 
-void DatabaseMemory::removeDataPath(ContextPtr local_context)
+void DatabaseMemory::removeDataPath()
 {
-    std::filesystem::remove_all(local_context->getPath() + data_path);
+    /// Remove database directory with symlinks to the tables' data
+    assert(TSA_SUPPRESS_WARNING_FOR_READ(tables).empty());
+    try
+    {
+        fs::remove_all(path_to_table_symlinks);
```

Comment:
Should we follow symlinks/mappings and delete all the data in the `store/` directory?

Reply:
No, `IStorage` is responsible for removing it on DROP TABLE. And we don't know the UUIDs after restarting the server (but we have unused dirs cleanup in `DatabaseCatalog`)